contracts/libraries/BisectionLib.sol

```diff
@@ -51,11 +52,11 @@ function bisection(
         root = (lower + upper) / 2;
 
         int256 output = fx(args, root);
-        if (output == 0) break; // Found the root.
+        lowerOutput = fx(args, lower); // Lower point could have changed in the previous iteration.
```

Comment:
I don't think this one is needed as we set it in the `else` branch?

Reply:
I think you're right, let's remove it.